src/fun_lti_provider/models.py

```diff
@@ -0,0 +1,120 @@
+    Model representing a LTI Consumer.
+    """
+
+    slug = models.SlugField(
+        verbose_name=_("consumer site identifier"),
+        primary_key=True,
+        help_text=_("identifier for the consumer site"),
+    )
+
+    title = models.CharField(
+        max_length=255,
+        verbose_name=_("Title"),
+        help_text=_("human readable title, to describe the LTI consumer"),
+        blank=False,
+    )
+
+    class Meta:
+        """Options for the ``LTIConsumer`` model."""
+
+        db_table = "lti_consumer"
+        verbose_name = _("LTI consumer")
+        verbose_name_plural = _("LTI consumers")
+
+    def __str__(self):
+        """Get the string representation of an instance."""
+        return self.title
+
+
+class LTIPassport(models.Model):
+    """
+    Model representing an LTI passport for LTI consumers to interact with the django application.
+
+    A LTI consumer can have multiple passports.
+
+    An LTI passport stores credentials that can be used by an LTI consumer to interact with
+    the django application acting as an LTI provider.
+    """
+
+    consumer = models.ForeignKey(LTIConsumer, on_delete=models.CASCADE)
+
+    title = models.CharField(
+        max_length=255,
+        verbose_name=_("Title"),
+        help_text=_(
+            "human readable title, to describe this LTI passport (i.e. : who will use it ?)"
+        ),
+        blank=False,
+    )
+
+    oauth_consumer_key = models.CharField(
+        max_length=255,
+        verbose_name=_("oauth consumer key"),
+        unique=True,
+        help_text=_(
+            "oauth consumer key to authenticate an LTI consumer on the LTI provider"
+        ),
+        editable=False,
+    )
+    shared_secret = models.CharField(
+        max_length=255,
+        verbose_name=_("shared secret"),
+        help_text=_("LTI Shared secret"),
+        editable=False,
+    )
+    is_enabled = models.BooleanField(
+        verbose_name=_("is enabled"),
+        help_text=_("whether the passport is enabled"),
+        default=True,
+    )
+
+    class Meta:
+        """Options for the ``LTIPassport`` model."""
+
+        db_table = "lti_passport"
+        verbose_name = _("LTI passport")
+        verbose_name_plural = _("LTI passports")
+
+    def __str__(self):
+        """Get the string representation of an instance."""
+        return self.oauth_consumer_key
+
+    # pylint: disable=arguments-differ
+    def save(self, *args, **kwargs):
+        """Generate the oauth consumer key and shared secret randomly upon creation.
+
+        Parameters
+        ----------
+        args : list
+            Passed onto parent's `save` method
+        kwargs: dict
+            Passed onto parent's `save` method
+
+        """
+        self.full_clean()
+        if not self.oauth_consumer_key:
+            self.oauth_consumer_key = "".join(
+                random.choice(OAUTH_CONSUMER_KEY_CHARS)
+                for _ in range(OAUTH_CONSUMER_KEY_SIZE)
+            )
```

Comment:
I would suggest to define a separated private method that generates this default key and avoid defining global variables that are only used here. Define them locally instead. Defining a method for this will allow to unit test the random key generation.

Reply:
I made some modifications after your comment : 
 - The consumer_key and secret key generation are now made inside dedicated methods (`generate_consumer_key` and `generate_shared_secret`)
 - I added tests in [test_models.py](https://github.com/openfun/ashley/blob/fun_lti_provider/tests/fun_lti_provider/test_models.py)
- I replaced the usage of the `random` module with `secrets` to have a stronger random generator and to fix bandit warnings.
- I added entropy in the length of the generated secrets and consumer keys

Reply:
It would be nice if you can port it to [marsha](https://github.com/openfun/marsha) so we can add bandit there as well.

Reply:
:+1: it's noted on my todo list.
I see that `marsha` is using python >= 3.6 so it won't be a problem to use `secrets`.